File: frontend-user-web/src/types/region.ts
/**
 * Phase 5 · 地区（省 / 市 / 区）类型。
 *
 * 严格对齐 docs/API/phase-5-contracts.md §3.5 / §7。
 */

export type RegionLevel = 1 | 2 | 3;

/** 单个节点。 */
export interface RegionOut {
  code: string;
  parent_code: string | null;
  name: string;
  short_name: string | null;
  level: RegionLevel;
  sort_order: number;
}

/** 树形节点（返回全量树时使用）。 */
export interface RegionTreeNode extends RegionOut {
  children: RegionTreeNode[];
}

File: backend/app/schemas/region.py
"""Region schemas — Phase 5 contract §7."""

from __future__ import annotations

from pydantic import BaseModel, ConfigDict


class RegionOut(BaseModel):
    """A single administrative-division row."""

    model_config = ConfigDict(from_attributes=True)

    code: str
    parent_code: str | None = None
    name: str
    short_name: str | None = None
    level: int
    sort_order: int


class RegionTreeNode(BaseModel):
    """Tree-shaped Region node used by ``GET /regions/tree``."""

    code: str
    parent_code: str | None = None
    name: str
    short_name: str | None = None
    level: int
    sort_order: int
    children: list[RegionTreeNode] = []


RegionTreeNode.model_rebuild()

__all__ = ["RegionOut", "RegionTreeNode"]
